code/client/src/components/AddressInput.jsx

```diff
@@ -0,0 +1,103 @@
+import { SearchOutlined } from '@ant-design/icons'
+import { Select } from 'antd'
+import React, { useCallback } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import walletActions from '../state/modules/wallet/actions'
+import util, { useWindowDimensions } from '../util'
+
+/**
+ * Renders address input that provides type ahead search for any known ONE addresses.
+ * Known addresses are addresses that have been entered by user for at least once.
+ * If an address is entered in none one1... format, the address will be converted and saved as one1... format.
+ */
+const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSelectOptions, knownAddressKey }) => {
+  const dispatch = useDispatch()
+
+  const wallets = useSelector(state => Object.keys(state.wallet.wallets).map((k) => state.wallet.wallets[k]))
+
+  const knownAddresses = useSelector(state => state.wallet.knownAddresses)
+
+  const network = useSelector(state => state.wallet.network)
+
+  const { isMobile } = useWindowDimensions()
+
+  const onChangeAddress = useCallback((address) => {
+    setAddressCallback(address)
+  }, [setAddressCallback])
+
+  /**
+   * Determines if the input wallet wallet is for the current wallet. Applicable for existing wallet management.
+   */
+  const notCurrentWallet = useCallback((inputWalletAddress) =>
+    !currentWallet || currentWallet?.address !== inputWalletAddress,
+  [currentWallet])
+
+  /**
+   * This is interim wallet object that uses known wallet addresses to be used to render the address selection with
+   * existing wallet object. The address key is unique as knownOneAddress. Note that this will not be saved as actual wallet.
+   */
+  const knownWallets = (knownAddresses ? knownAddresses[knownAddressKey] || [] : []).map((knownOneAddress) => ({
+    network,
```

Comment:
see below comments on network - the network should be retrieved from state, based on which network was selected when the address was added. The network should not be just added based on the current selected network right now

Reply:
Added the network to known address object and used as filter when display them.